arbos/block_processor.go

```diff
@@ -193,6 +194,21 @@ func ProduceBlockAdvanced(
 
 	header := createNewHeader(lastBlockHeader, l1Info, arbState, chainConfig)
 	signer := types.MakeSigner(chainConfig, header.Number, header.Time)
+
+	if tracer != nil {
+		tracer.OnBlockStart(
```

Comment:
So I am not sure if this is me misunderstanding the intended API, but my impression from reading the code samples in the geth documentation is that a tracer does not need to implement every hook, only the ones it needs e.g. 

```golang
return &tracing.Hooks{
	OnTxEnd:      t.OnTxEnd,
}, nil
``` 
instead of writing hooks for every possible hook and then having no-ops for the hooks you don't care about.

If you try to write a tracer without OnBlockStart and OnBlockEnd here you get a null pointer derefence 

Also it appears none of the other hooks are called? I defined an OnTxEnd hook but did not see it get called anywhere - maybe that's my implementation error?